pkg/gatherers/common/gather_logs.go

```diff
@@ -79,23 +80,33 @@ func CollectLogsFromContainers( //nolint:gocyclo
 	var skippedContainers int
 	var records []record.Record
 
+	var podNameRegex *regexp.Regexp
+	if len(containersFilter.PodNameRegexFilter) > 0 {
```

Comment:
I think this should be validated/enforced by the JSON schema validation or? In other words we require some identificator of a Pod name.

Reply:
Not sure If I got it right. I just added this validation to not compile the regex with `nil` value... maybe there is no difference? Or are you saying that this should be required?

Reply:
Yes I think it should be required.

Reply:
OK, but... If we make it required it would cause some issues to existing log gatherers.

Reply:
We can't make it required, if we do we can't move the current log gathers to the conditional, but, on the other hand, we can make it required at the conditional scheme if we decide to not migrate the existing log gatherers to the condition or change them.